File: frontend/src/types/product.ts
export type Product = {
  id: string
  productVariantId?: string
  name: string
  slug: string
  description: string
  category: string
  brand?: string
  status?: string
  sku?: string
  color?: string
  size?: string
  price: number
  salePrice?: number | null
  stock: number
  quantityOnHand?: number
  quantityReserved?: number
  quantityAvailable?: number
  lowStockThreshold?: number
  isActive?: boolean
  image: string
  images?: string[]
  concurrencyStamp?: string
  rating: number
  tags: string[]
  imageFit?: 'cover' | 'contain'
  imageSurfaceClassName?: string
  imagePositionClassName?: string
}

File: backend/Booking.Domain/Entities/Product.cs
namespace Booking.Domain.Entities;

public sealed class Product
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public string Status { get; set; } = "Draft";
    public decimal Price { get; set; }
    public decimal? SalePrice { get; set; }
    public int StockQuantity { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAtUtc { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public string ConcurrencyStamp { get; set; } = Guid.NewGuid().ToString("N");
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}
